src/hooks/Orderbook/useOrderbookValues.ts

```diff
@@ -23,27 +23,33 @@ export const useCalculateOrderbookData = ({ maxRowsPerSide }: { maxRowsPerSide:
       orderbook?.asks?.toArray() ?? []
     )
       .map(
-        (row: OrderbookLine, idx: number) =>
-          ({
-            key: `ask-${idx}`,
-            side: 'ask',
-            mine: subaccountOrderSizeBySideAndPrice[OrderSide.SELL]?.[row.price],
-            ...row,
-          }) as PerpetualMarketOrderbookLevel
+        (row: OrderbookLine, idx: number): PerpetualMarketOrderbookLevel =>
+          Object.assign(
+            {},
+            {
+              key: `ask-${idx}`,
+              side: 'ask' as const,
+              mine: subaccountOrderSizeBySideAndPrice[OrderSide.SELL]?.[row.price],
+            },
+            row
+          )
       )
       .slice(0, maxRowsPerSide);
 
     const bids: Array<PerpetualMarketOrderbookLevel | undefined> = (
       orderbook?.bids?.toArray() ?? []
     )
       .map(
-        (row: OrderbookLine, idx: number) =>
-          ({
-            key: `bid-${idx}`,
-            side: 'bid',
-            mine: subaccountOrderSizeBySideAndPrice[OrderSide.BUY]?.[row.price],
-            ...row,
-          }) as PerpetualMarketOrderbookLevel
+        (row: OrderbookLine, idx: number): PerpetualMarketOrderbookLevel =>
+          Object.assign(
```

Comment:
These kotlin multiplatform types with getters don't play nice with typescript which thinks they may be on the prototype instead of base object so we have to use object.assign to get correct types.